ratelimits/names_test.go

```diff
@@ -232,3 +232,111 @@ func TestValidateIdForName(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildBucketKey(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          Name
+		desc          string
+		regID         int64
+		domain        string
+		domains       string
+		ipAddr        string
+		expectErr     string
+		bucketKeyTest func(t *testing.T, key string)
+	}{
+		{
+			name:   NewRegistrationsPerIPAddress,
+			desc:   "valid IPv4",
+			ipAddr: "1.2.3.4",
+			bucketKeyTest: func(t *testing.T, key string) {
+				test.AssertEquals(t, fmt.Sprintf("%d:1.2.3.4", NewRegistrationsPerIPAddress), key)
+			},
+		},
+		{
+			name:   NewRegistrationsPerIPv6Range,
+			desc:   "valid IPv6",
+			ipAddr: "2001:db8:abcd:0012::1",
```

Comment:
Tiny nit: RFC 5952 representation would be `12` instead of `0012`. That doesn't actually matter in this context, I'm just being very pedantic about IPv6 this month.